DiamondLightSource/python-artemis
# 413 systematize callbacks

Fixes #413 

Reorganises the external interaction submodule so that it can better be extended for different experiment types (eg #368 #369 #370)

### To test:
1. check the structure for sanity
2. run tests
